ggml-metal.metal

```diff
@@ -528,42 +535,102 @@ kernel void kernel_mul_mat_f16_f32(
     device const half  * x = (device const half  *) (src0 + r0*nb01 + im/(ne12/ne02)*nb02);
     device const float * y = (device const float *) (src1 + r1*nb11 + im*nb12);
 
-    uint ith = tpitg.x;
-    uint nth = tptg.x;
+    float sumf = 0;
+    if (ne00 < 128) {
```

Comment:
Should we have 2 separate kernels to avoid this branch?

Reply:
I was considering it, but taking into account the desire for shorter and simple code didn't do it. This is something one needs to study more carefully anyway. The best way to perform this computation is not just a function of `ne00`.